bindings/node/lib/autoEncrypter.js

```diff
@@ -115,13 +117,9 @@ module.exports = function (modules) {
           : this._bson.serialize(options.encryptedFieldsMap);
       }
 
-      if (options.kmsProviders) {
-        mongoCryptOptions.kmsProviders = !Buffer.isBuffer(options.kmsProviders)
-          ? this._bson.serialize(options.kmsProviders)
-          : options.kmsProviders;
-      } else if (!options.onKmsProviderRefresh) {
-        throw new TypeError('Need to specify either kmsProviders ahead of time or when requested');
-      }
+      mongoCryptOptions.kmsProviders = !Buffer.isBuffer(this._kmsProviders)
```

Comment:
No longer throw an error if `kmsProviders` are empty since we can auto-refresh from the environment.